Spark Configuration Api's Swagger Changes

Bharath550-ms opened the pull request:
Spark Configuration Api's Swagger Changes

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Bharath550-ms committed:
Spark Configuration Api's Swagger Changes

Spark Configuration Api's Swagger Changes

Bharath550-ms committed:
Spark Configuration Api's Swagger Changes

Spark Configuration Api's Swagger Changes

Bharath550-ms committed:
Added Spark Configuration Api's for stable versions

Added Spark Configuration Api's for stable versions

Bharath550-ms committed:
Added Spark Configuration Api's for stable versions

Added Spark Configuration Api's for stable versions

ajpeacock0 committed:
Rename CommunicationTurnCredentialsResponse to CommunicationRelayConfiguration (#14762)

muazmian committed:
Adding totalCount to transactionList result (#14858)

msyyc committed:
[python] readme.python.md configuration for `monitor` and `network` (#14968)

* monitor

* network

* monitor

* monitor

chenjianfei2017 committed:
add tag to readme.go.md (#14967)

* add tag to readme.go.md

* Update specification/azurestackhci/resource-manager/readme.go.md

Co-authored-by: Arcturus <dapzhang@microsoft.com>

Co-authored-by: Arcturus <dapzhang@microsoft.com>

msyyc committed:
python config (#14984)

Bharath550-ms committed:
Removing required field for notes attribute

Removing required field for notes attribute

Bharath550-ms committed:
Removing required field for notes attribute

Removing required field for notes attribute

Bharath550-ms committed:
Updated changes

Updated changes

Bharath550-ms committed:
Updated changes

Updated changes

msyyc committed:
[Python] track2 config  (#14990)

* authorization

* automation

* recoveryservices

* logic

* notificationhubs

* search

Bharath550-ms committed:
Updated Required fields for SparkConfigurationInfo Object

Updated Required fields for SparkConfigurationInfo Object

Bharath550-ms committed:
Updated Required fields for SparkConfigurationInfo Object

Updated Required fields for SparkConfigurationInfo Object

msyyc committed:
alertsmanagement,applicationinsights,attestation,batch,cost-management,netapp,synapse (#15000)

raviten committed:
[DataBox] Bump the api version to add new values for enums for JobStageStatus (#14909)

* copy of 2021-03-01, first commit

* adding new api version and its example

* updating readme files with 2021-05-01 version and cSpell.json too with Tera as acceptable word

* updating 2021-readme files

* adding reference to the file

* removing track1 changes for python

msyyc committed:
track2 config (#15004)

msyyc committed:
[python]track2 config (#15001)

* track2 config

* directive for required type in readme.python.md

* Update readme.md

* Update readme.python.md

* Update readme.python.md

jeremymeng committed:
[ContainerRegistry] update for better codegen for data plane Track 2 SDK libraries (#13711)

* [Track2 Data Plane SDK] prepare folder

Add a new folder Azure.ContainerRegistry with a copy of
preview/2019-08-15 specs in preparation of further changes to make
better codegen of Track 2 data plane SDK libraries

* Add Track 2 customizations

* update PATCH return types

* Add a preliminary readme.md

* Fix CI Avocado failures

- rename folder to be consistent with api version
- remove GetAccessToken.json example as the endpoint was removed in this swagger

* Add timeasc and timedesc to custom-words.txt

also remove sections unrelated to Track 2 SDK

* Update typescript customization

* Fix required-ness for some properties

Fix casing aadAccesstoken => aadAccessToken

* Run prettier

* updates to swagger for Beta 2 API

* bug fix in arch enum

* updates per apiview feedback

* updates to manifest properties

* Move some customization into swagger

* Undo client name customization for `architecture`

Now that we use `architecture` in the new API design.

* Update arch and os types and set readOnly: true

* make change for grant type = password

* editing the correct swagger

* undoing changes to the wrong swagger

* adding internal enum for grant type

* moving grant_type to the end of the parameters for .net

* Incorporate Anne's beta.3 swagger update

and make ManifestAttributes.manifest required so its flattened
`digest` property can have their proper required-ness.

* small name changes

* updates to separate out ArtifactManifestReference

* flatten writeable property types; add registry property back

* update the delete methods

* Remove property required-ness

as they are used for input and cannot be used together with "readOnly"

* Run prettier

* Fix linting errors in examples

* Remove extraneous brace

* Add custom word for ACR

* Fix more linting errors in examples

* Revert "Remove property required-ness"

This reverts commit 4eba58a49bfd3a3e51219c90f583521391596218.

* Remove default value of "none" for architecture enum

* renames per arch board review

* Copy the existing suppression from master

* Fix spellcheck errors

Add words to custom-words.txt

* More spelling fix

* Suppress R2056 RequiredReadOnlyProperties

Reasoning at https://github.com/Azure/azure-openapi-validator/issues/243

* Remove `:` in reason value

it breaks yaml

* Update description of registryLoginServer

as `azurecr.io` is kust for public cloud and the format can be
different for every cloud or vanity URLs.

* Address PR feedback

- Add "Azure" to title
- add `"teleportEnabled": false` to example

* Minor update to tyepscript codegen customization section

Co-authored-by: Anne Loomis Thompson <annelo@microsoft.com>
Co-authored-by: seankane-msft <seankane@microsoft.com>
Co-authored-by: Pallavi Taneja <pallavit@users.noreply.github.com>

ajpeacock0 committed:
Adding 2021-06-21 TURN API version (#14955)

herreraj-ms committed:
Microsoft.Insights - Fix .NET SDK generation (#14998)

* Microsoft.Insights - Fix .Net SDK generation

* adding directive to tag package-2021-04

msyyc committed:
[python]track2 config for some services (#15007)

* consumption

* delete track1 config

DaeunYim committed:
typescript readme fixed (#14985)

Commit:
Add properties pipelineName and pipelineRunId. (#14914)

Co-authored-by: Vitor da Silva <t-vdasilva@microsoft.com>

larisaborodina committed:
[AzureStack] [Admin API] ACR Admin swagger spec changes (#14290)

* ACR Admin swagger spec changes

* Release/larisab (#1)

* Add swagger spec for acr setup.

* Add swagger spec for acr setup - fixed jsons.

* Add swagger spec for acr setup - address feedback to convert get status to post.

* Add swagger spec for acr setup - fixed typo.

* Add swagger spec for acr setup - added setup specs to readme.

* Add swagger spec for acr setup - fixed checkSetup definitions.

* Add swagger spec for acr setup - addressed comments.

* Add swagger spec for acr setup - addressed ARM feedback.

* Add swagger spec for acr setup - added proper input-files n readme.

* Add swagger spec for acr setup - fixed inavalid scheme.

* Add swagger spec for acr setup - fixed inavalid method Post instaed of Get.

Co-authored-by: Larysa Borodina <larisab@microsoft.com>

* Update API changes (#2)

* Update API changes

* Add delete call implementation

* update examples

* Fix swagger validation errors. (#3)

* Fix swagger validation errors - 2-nd. (#4)

* Fix model validation fix

* Fix swagger validation errors - 3d. (#5)

* Fix swagger validation errors - 4th. (#6)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Release/acrchanges (#8)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Release/acrchanges (#9)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Release/larisab (#11)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 12th.

* Release/acrchanges (#12)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Release/acrchanges (#13)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Release/acrchanges (#14)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Release/acrchanges (#15)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Release/acrchanges (#16)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Release/acrchanges (#17)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Release/acrchanges (#18)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Release/acrchanges (#19)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Release/acrchanges (#20)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Release/acrchanges (#21)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Release/acrchanges (#22)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Release/acrchanges (#23)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Fix swagger specs - 34th

* Release/acrchanges (#24)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Fix swagger specs - 34th

* Test cmdlet name creation with recommended operation id for quota

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Release/acrchanges (#25)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Fix swagger specs - 34th

* Test cmdlet name creation with recommended operation id for quota

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Use v2 common resource types.

* Release/acrchanges (#26)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Fix swagger specs - 34th

* Test cmdlet name creation with recommended operation id for quota

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Use v2 common resource types.

* Use v2 common resource types for parameters also.

* Release/acrchanges (#27)

* Fix swagger validation errors - 4th.

* Fix swagger validation errors - 5th.

* Fix swagger validation errors - 6th.

* Fix swagger validation errors - 7th.

* Fix swagger validation errors - 13th.

* Fix swagger validation errors - 14th.

* Fix swagger validation errors - 15th.

* Fix swagger validation errors - 16th.

* Fix swagger validation errors - 17th.

* Fix swagger validation errors - 18th.

* Fix swagger validation errors - 19th.

* Fix swagger validation errors - 20th.

* Fix swagger validation errors - 21th.

* Fix swagger validation errors - 21st.

* Fix swagger validation errors - 22st.

* Fix swagger validation errors - 23st.

* Fix swagger validation errors - 24st.

* Fix swagger validation errors - 25th.

* Fix swagger validation errors - 26th

* Fix swagger validation errors - 27th

* Fix swagger validation errors - 28th

* Fix swagger validation errors - 29th

* Fix swagger validation errors - 30th

* Fix swagger validation errors - 31th

* Fix swagger specs, address comments and naming bug - 32th

* Fix swagger specs, address comments and naming bug - 33th

* Fix swagger specs - 34th

* Test cmdlet name creation with recommended operation id for quota

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Test cmdlet name creation with recommended operation id

* Use v2 common resource types.

* Use v2 common resource types for parameters also.

* Use v2 common resource types for parameters also. - fixed the readme to remove unneeded file

Co-authored-by: Rohit Jaini <rjaini@microsoft.com>
Co-authored-by: larysaborodina <larysaborodina@gmail.com>

skayani committed:
[ManagedServices] Update property descriptions before new API docs are published (#14881)

* Update property descriptions before new API docs are published

* ID -> identifier and specify AAD principal in principalId descriptions

* Addressed feedback from Jenny

* Addressed feedback from Ramu

* Spell check fix

* Semantic fixes

matthew-ouellette committed:
Update SecurityInsights metadata preview API (#15012)

* Copy commit from private repo

* Make new API version

* Remove old examples

* Fix exclusion for Metadata list endpoint

* Move back to initial preview version

* Change id to contentId, sourceId

* Fix sourceId in get all example

* Update enums, required fields

* Update enums, required fields

weidongxu-microsoft committed:
limit the ResourceGroupNameParameter directive to package-2021-04 (#15019)

idear1203 committed:
[Synapse] - Update sql pool v3 specs (#14839)

* [Synapse] - Update sql pool v3 specs

* Address comments

Co-authored-by: Dongwei Wang <dongwwa@microsoft.com>

chenjianfei2017 committed:
add tag into readme.go.md (#15022)

jsedlak-microsoft committed:
Update Mobility retirement text (#14980)

* Update Mobility retirement text

Link to retirement details

* Change banner tags

* Banner take 3

* Address Spelling issues with API's

* Adjust banner

* Undo edits for unchanged lines

* Unchange lines

chcmsft committed:
update location url of post async batch api (#14965)

pilor commented:
Existing API versions should not be changed.  The new APIs should only go into a new API version.  If you would like to request an exception please follow the Breaking Change Process outlined in the PR description

pilor commented on specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-05-01/sparkConfiguration.json:
```diff
@@ -0,0 +1,214 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sparkconfigurations/{sparkConfigurationName}": {
```
This new resource type only has a GET exposed.  How is it created?

Bharath550-ms replied:
Yes, We need just GET operations here to attach spark configurations while updationg the SparkPool

bleroy committed:
New search 2021-04-30-preview dataplane version (#14919)

* Create new version folder without changes.

* Update version number

* Update readme with new version information

* Add the identity property to data source.

* Add captions and semantic fields

* Move the new captions and semantic fields parameters after existing ones to preserve order.

* OCR skill line ending parameter

* Add languages to translate skill (includes additional word for Klingon language code)

* Add identity to encryption key type.

* Fix doc for new CR line separator

renuka-raju committed:
Peering api 2021-06-01 (#14703)

* Existing version in new directory

* 2021-06-01 swagger correctness changes

* Resolved conflict

* [feat] Add looking glass documentation to 2021-06-01

* [fix] Change LookingGlasses to LookingGlass

* Add connection monitor models and examples

* Remove LookingGlass response status codes and fix parameter capitalization

* Fix ReadOnly properties in request

* Remove last readOnly property

Co-authored-by: Renuka Raju <rraju@microsoft.com>
Co-authored-by: Jimmy Xu <t-jimmyxu@microsoft.com>
Co-authored-by: Slava Uryumtsev <suryumtsev@microsoft.com>

ArcturusZhang committed:
Introduce go SDK track 2 configurations for monitor (#14588)

* introduce monitor track 2 configurations

* add directive to solve the duplicate schema error

* update package name

* append a comment for the reason of directive

vakuncha committed:
moving swagger to stable (#14992)

* Base Swagger version as per guideline

* Removing preview from directory version

* changing version name in all files

* updating readme and go read me

* supressing ingestion keyin read me

ToddKingMSFT committed:
Update readme.md (#14933)

Add R4007 suppression for metricNamespaces since changing it's return type would be a breaking change

giakas committed:
[Hub Generated] Review request for Microsoft.Media to add version stable/2021-06-01  Part-1 (#14970)

* Adds base for updating Microsoft.Media from version stable/2021-05-01 to version 2021-06-01

* Updates readme

* Updates API version in new specs and examples

* Update the branch with copy of all other services 2020-05-01 version files except encoding.

* update AMS to 2021-06-01 Part1 (All swaggers except encoding)

* fix some lint errors.

* Fix an unreferenced file.

* fix more linting errors

* remove some unreferenced files.

* Fix systemData in streamingservice

* fix more linting errors

* fix 404s

* fix more lint errors

* fix 204 issue for delete

* fix definitions to use v2 common-types

* remove type for the live event list list defenition

* Revert "remove type for the live event list list defenition"

This reverts commit 0f66253d284284037bf2c0d7a04c947d4dca7715.

* removed unused definition

zhenglaizhang committed:
Bump dependencies (#14987)

* bump prettier

* bump jsonpath

* remove rest-api-specs-scripts

* fix prettier-fix command

liranc committed:
Added new onboarding modes and optinal-readonly evaluationEndTime property (#15023)

Co-authored-by: Liran Chen <lirche@microsoft.com>

tofriedl committed:
add all apis to 2020 readme (#14870)

* add all apis to 2020 readme

* fix

* add 2020 to Go Readme

* change error definitions in 2020-02-02

* changes to fix issues

* fix more errors

* change error response to custom to fix errors

* add code property to error

* fix error response

* change errorResponse name

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* fixes

* fixes

* fix

* fixes

* fix remaining issues in readme.go.md

Co-authored-by: ArcturusZhang <dapzhang@microsoft.com>

diwudd committed:
Microsoft.Features: Add new API version 2021-07-01 (#14736)

* The previous version in the 2021-08-1 folder

* Add new feature registration apis

* Update readme files

* Update readme file

* Remove feature configuration apis

* Fix avocado

* Fix validations

* Fix validation

* Fix validations

* Fix validation

* Fix title

* Fix validation

* Fix validation

* Fix readme

* Fix validation

* Fix validation

* Fix validation

* Fix validation

* Fix validation

* suppress r4009

* Fix validation

* Fix validation

* Fix python

* Fix readme

* Fix python

* Remove post api

* Update example api-versions

Co-authored-by: shenshengkafei <wudiadi1111@gmail.com>

paterasMSFT committed:
Batch DataPlane 2021-06-01.14 update (#14830)

* Copy of 2020-09-01 for ease of review

* Azure Batch updated to 2021-06-01.14

Suppressions

Fixed property name.

Fixed type

Fixed casing

Fixed casing

Spelling fix

Fixed enums

Fixed enum

Fixed EOF and casing from PR feedback

Markdown fixes

Added Managed Identity and Authentication Mode APIs.

Fixed example

Spot cleanup

Casing fix

Fixed spacing

Added maxParallelTasks.

Formatting.

Removed newline

Reverted markdown change to satisfy Avocado

http://azure.github.io/autorest/user/literate-file-formats/configuration.html#the-file-format

roipmsft committed:
Move RestorePoint properties into new properties bag for restore point (#14638)

jlichwa committed:
Remove files from package which are not part of management plane SDK - update readme.md (#15047)

* Update readme.md

* Update readme.md

asmaskar committed:
fix in readme file for latest sdk generation (#15037)

RAY-316 committed:
T2 elastic (#15049)

* sql t2 readme config

* readme config

* Update readme.python.md

* conflient resolve

* readme config

Co-authored-by: msyyc <70930885+msyyc@users.noreply.github.com>

arv100kri committed:
[Azure Cognitive Search] Fix typos and add some descriptins (#14903)

* Fix typos and add some descriptins

* more description changes

Co-authored-by: Arvind Krishnaa Jagannathan <arjagann@microsoft.com>

zhaomuzhi committed:
Remove WorkspaceConnectionDto and use WorkspaceConnection instead for PUT request (#14705)

* Add Identity to createWorkpace example

* change for getTransitivePEUsages

* update

* prettier code

* add another new endpoint

* address comments

* change for getTransitivePEUsages

* update

* prettier code

* add another new endpoint

* address comments

* revert change on transitivePE and notebookKeys

* revert change on transitivePE and notebookKeys

* add new stable version 2021-01-01

* add ListNotebookAccessToken

* add primaryUserAssignedIdentity

* fix missing properties and apis

* fix one more missing property

* 202 status for start and stop CI

* make resync keys long runnin operation in version Jan2021

* resync keys 204

* update resync keys example

* Dont throw 204 for resync

* update resync keys

* Formatting

* revert Jan2021changes and add new version

* start apr2021 version for azure ml rp

* undo remove new line in jan2021

* use systemData definition in common

* removed old files

* fix small swagger bugs

* run prettier

* remove reference to sku

* update readme

* minimum agentCount should be 0

* add createdOn and ModifiedOn to compute properties

* add LoadBalancerType to aks compute

* add isNotebookInstanceCompute to virtualMachine

* add leafDomainLabel to SslConfiguration

* add overwriteExistingDomain to SslConfiguration

* add synapseSpark as supported compute-type

* update patch compute response to 202

* remove location from example responses

* Revert "remove location from example responses"

This reverts commit 048bf2551513afae842a9c4fbb5861aeddc6c33a.

* MachineLearningCompute_Update should have a 200 response

* sys data to use local def due to naming conflicts.

* Fix enum casing

* Fix operationId PR comment.

* Change opId in apr version

* remove 20210401 for easy review

* only change api version

* merge changes for 20210401 version

* Add x-ms-discriminator-value for Synapse spark

* camelcase loadbalancer properties

* add createdOn and Modified on to examples

* address AvoidAnonymousTypes error for SynapseSpark

* Minor fixes + new property

* Remove subscription state.

* fix swagger bug

* fix example

* fix more reported problems

* remove nullable flag for tags

* make containerRegistry nullable

* address comments

* fix example

Co-authored-by: Vinay Rajmal Soni <vison@microsoft.com>
Co-authored-by: Suba <sukaruna@microsoft.com>
Co-authored-by: jopet@microsoft.com <jopet@microsoft.com>
Co-authored-by: Josh Peterson <15050193+joshanator@users.noreply.github.com>
Co-authored-by: Suba <sk.suba1@gmail.com>

kaiqzhan committed:
AKS: replace CloudError with ManagedClusterPodIdentityProvisioningError for AAD pod identity provisioning spec (#15054)

Co-authored-by: Kaiqi Zhang <kaiqzhan@microsoft.com>

paterasMSFT committed:
Batch Control Plane 2021-06-01.14 update (#14862)

* Copy commit for code review

* Initial changes

Prettier

Suppressions

Added Managed Identity and Authenticate Mode APIs

Added type

Replace spot.

Added line break

Newline

Revert "Newline"

This reverts commit 735a19497174fcbc1cb2c7d7835205018614eb42.

Revert "Added line break"

This reverts commit bd0ceca43016e7df20bbc1280271af0f4b503b8d.

Reverted markdown change to satisfy Avocado

http://azure.github.io/autorest/user/literate-file-formats/configuration.html#the-file-format

msyyc committed:
python track2 config (#15050)

* python track2 config

* redis

* redis

k-rush committed:
Microsoft.TimeSeriesInsights: Fix swagger correctness issues (#14469)

* Fix swagger correctness issues

* Fix casing in localTimestamp

* Fix localTimeStamp property

* Fix lintdiff errors

ahrkrak committed:
Remove CoreOS, add Flatcar (#14796)

dw511214992 committed:
doc example bug fix (#14754)

dw511214992 committed:
change profile to profile-content in readme.typescript.md (#15034)

dw511214992 committed:
change main branch to feature/v4  in js sdk (#15046)

vikramkotturu committed:
Users/vikottur/ws02 datasource auth (#14932)

* Created base 2021-06-01-preview a copy of 2021-02-01-preview.

* Added my changes to support datasource auth credentials ontop of the base from previous version.

* Fixed as issue with FindRestorableTimeRanges.json example.

t-vjanardhan committed:
[EdgeOrder] Update 2020-12-01-preview swagger and add readme files (#14935)

* new api changes (#1)

* new api changes

* referer url change

* EdgeOrder 2020-12-01-preview Swagger changes (#5)

* readme changes

* Swagger refresh and update examples

* Change examples as per oav validations

* Minor swagger modifications

* Minor spacing adjustments

* Set ModelAsString to true for enums and modify examples

* Fix swagger lint validation error

* Modify readme and examples

* Modify readme.go.md

* Update swagger

* Minor prettier fix

* Change python package version in readme

* Fix spell check error

Co-authored-by: sashanm <54701252+sashanm@users.noreply.github.com>

shinilm committed:
Changes to fix the Resource Provider name casing issue for Azure Postgres, MySQL and MariaDB  (#15016)

* Changes to fix the Resource provided name casing issue for Azure Postgres, MySQL and MariaDB services

* Changes to fix the Resource provided name casing issue for Azure Postgres, MySQL and MariaDB services

* Revert "Changes to fix the Resource provided name casing issue for Azure Postgres, MySQL and MariaDB services"

This reverts commit ecc38dc560b77de750350143fd80bdd6e3823882.

Sai-Kumar-1901 committed:
Added missing properties for managed Private endpoints (#15014)

* Added missing properties for managed Private endpoints

Added missing properties for managed private endpoints in stable version.
Added the missing swagger files for managed private endpoints in the latest  preview version.

* fixed indentation issues

* fix indentation issues1

* ran prettier-fix

* Added missing examples files

* fixed examples

* fixed version in preview swagger

* Update readme.md

Updated readme.md

* Update readme.md

Co-authored-by: Sai-Kumar-1901 <saikumar@microsoft.com>

SayanMS09 committed:
New api version support to include disable local auth flag for event hub (#14950)

* New api version support to include disable local auth flag for event hub

* Updated api-version in examples

* Added auth examples in NameSpaces

* Fixing LintDiff error

* Fixing LintDiff error contd

* Removed UserassignedIdentities as part of definition

* Changed to UserAssignedIdentity as a reference in userAssignedIdentities object

Co-authored-by: Sayan Ghosh <sayanghosh@microsoft.com>

LeoVannelli committed:
S360 (#14362)

* Add dividePerInstance property to the autoscale metric trigger

* Update examples for autoscale to include 'dividePerInstance'

* Update examples for correctness

* update example for correctness

* swagger corrections

* remove entry from wrong tag

* fix boolean as string validation error

* validation fix

* added missing lifter changes

* add recommended linter suppression

avjerafi committed:
[Hub Generated] Review request for Microsoft.Insights to add version stable/2018-04-16 (#15072)

* Added suppression for R4005 linting error

* Using Pascal case for resource provider

* Added all missing type:object

* Added supperssion for R3016 - camel case for odata.type

Co-authored-by: Avi Jerafi <avjerafi@microsoft.com>

shih-che committed:
Add optional param - private DNS zone to MySQL server API (#14964)

* add optional parameter private dns zone to server create/update

* fix lintdiff errors

* fix prettier issues

RAY-316 committed:
T2 vmware (#15077)

* sql t2 readme config

* readme config

* Update readme.python.md

* conflient resolve

* readme t2 config

Co-authored-by: msyyc <70930885+msyyc@users.noreply.github.com>

amitelran committed:
[Hub Generated] Review request for Microsoft.Synapse to add version preview/2021-06-01-preview (#14814)

* Adds base for updating Microsoft.Synapse from version stable/2021-05-01 to version 2021-06-01-preview

* Updates readme

* Updates API version in new specs and examples

* Add kustoPool.json to new API version + examples files + in readme.md

* Add "type": "object" to definitions

* Modify SKU to new name & size instead of previous name & tier

* Add KustoPool stop

* fix KustoPool Stop example params

* Add KustoPool Start + example

* Add KustoPool LanguageExtension Add/List/Remove

* Add "type": "object" to LanguageExtensionsList

* Add KustoPoolFollowerDatabases DETACH and LIST + examples

* Add KustoPoolDatabasePrincipals ADD & LIST + examples

* Add missing DatabasePrincipalListRequest definition

* Add "type": "object" to FollowerDatabaseListResult, DatabasePrincipalListRequest, DatabasePrincipalListResult

* Add KustoPoolDatabasePrincipalsRemove + example

* Remove redundant newline

* Add AttachedDatabaseConfigurations to KustoPool + examples

* Add AttachedDatabaseConfigurationNameParameter

* Fixes for AttachedDatabaseConfiguration

* Add "type": "object" to TableLevelSharingProperties

* Fix example of KustoPoolAttachedDatabaseConfigurationCreateOrUpdate.json

* Add DataConnectionValidation to kustoPool.json + example

* Fixes to DataConnectionValidation

* Remove properties from KustoPoolDataConnectionValidation.json

* Fix object instead of array at KustoPoolDataConnectionValidation.json example

* Fix indentation at KustoPoolDataConnectionValidation.json

* Add KustoPoolPrincipalAssignmentCheckNameAvailability to kustoPool.json + example

* Remove redundant slash

* Add KustoPoolDatabasePrincipalAssignmentsCheckNameAvailability to kustoPool.json + example

* Fix in KustoPoolDatabasePrincipalAssignmentsCheckNameAvailability.json

* Add KustoPoolDataConnectionCheckNameAvailability.json + example

* Add KustoPoolListSkus + example

* Add kustoSkus path to kustoPool.json + example

* Remove {databaseName}/addPrincipals, listPrincipals, removePrincipals.
Remove related definitions + example files

* Change workspaceUid to workspaceUID to keep convention with Workspace resource

* Add new KustoPool properties:
optimizedAutoscale
enableStreamingIngest
enablePurge
languageExtensions

* Change "KustoCluster" to "KustoPool" and remove engineType property

* Change "kustoSkus" endpoint to "skus"

* Add KustoPool databases CheckNameAvailability + examples

* Re-position kustoPoolCheckNameAvailability

* Fix typo in example

* Minor adjustments in KustoPool databases CheckNameAvailability

* Change to "validateDataConnection"

* Revert back to DataConnectionValidation to keep consistency with kusto swagger

Co-authored-by: Amit Elran <amelran@microsoft.com>

brpanask committed:
[Hub Generated] Review request for Microsoft.StorageCache to add version stable/2021-03-01 (#15013)

* Address missing defaults noted by s360

* Update to remove default of array of objects that failed to validate.

lagalbra committed:
Microsoft.HanaOnAzure swagger: Update spec to add missing fields (#14879)

* Microsoft.HanaOnAzure swagger: Update spec with missing fields from RP

Add HanaSID array to StorageProfile properties

* Update ListOperations API to include isDataAction field

* Add examples for HanaSID field

* correct bool to boolean

* Fix LintDiff error introduced by this PR

idear1203 commented:
1. Please also update Swagger spec of the latest preview API version 2021-06-01-preview. We will use this version to build next SDK.
2. Is the API deployment at server side complete or not?

imshashwataggarwal committed:
[Hub Generated] [SiteRecovery] Review request for Microsoft.RecoveryServices to add version stable/2021-04-01 (#14930)

* Adds base for updating Microsoft.RecoveryServices from version stable/2021-02-10 to version 2021-04-01

* Updates readme

* Updates API version in new specs and examples

hassanchawiche committed:
Added Watchlists Stable Version 2021-04-01 (#14843)

* Copied Watchlists version 2021-03-01-preview as is to new directory structure

* Added Watchlist stable version 2021-04-01

nimengan committed:
[Iothub] Adding new properties in IothubProperties [stable version: 2021-07-01] (#14894)

* Adds base for updating Microsoft.Devices from version stable/2021-03-31 to version 2021-07-01

* Updates readme

* Updates API version in new specs and examples

* adding new properties in iothu properties

* Adding isVerified property in preview version

* validation fixes

moranraz committed:
extract AutomationRules to new file (#15075)

* extract AutomationRules to new file

* fix errors

* change ResourceWithEtag

* update readme

* change resource and subscriptionId

* add bracket

* .

* .

* .

* revert common definitons

* .

moranraz committed:
Stable version separation (#15076)

* AlertRules

* bookmarks

* dataConnectors

* add missing definitions

* fixes

* correct version

* .

* revert common definitions

* prettier

ArcturusZhang committed:
fix the directive to exclude changes for track 2 (#15084)

* fix the directive to exclude changes for track 2

* add model-deduplication

* fix

Bharath550-ms committed:
Merge branch 'AddSparkConfigurationArtifactApis' of https://github.com/Bharath550-ms/azure-rest-api-specs into AddSparkConfigurationArtifactApis
